text/0057-single-field-shortcut.md

```diff
@@ -0,0 +1,167 @@
+
+```python
+from amaranth import *
+from amaranth.lib import wiring
+from amaranth.lib.wiring import In, Out
+from amaranth_soc import csr
+
+class Toggle(wiring.Component):
+    bus: In(csr.Signature(addr_width=1, data_width=32))
+    toggle: Out(32)
+
+    def __init__(self):
+        # Register can take the Field directly, eliminating the subclass
+        self._toggle = csr.Register(csr.Field(csr.action.RW, 32), access="rw")
+
+        # ... bridge setup ...
+
+    def elaborate(self, platform):
+        m = Module()
+
+        # ... bridge setup ...
+
+        for i in range(32):
+            # FieldAction accessed directly, no need to specify `toggle` again
+            with m.If(self._toggle.f.data[i]):
+                m.d.sync += self.toggle[i].eq(~self.toggle[i])
+            with m.Else():
+                m.d.sync += self.toggle[i].eq(0)
+
+        return m
+```
+
+## Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+(with reference to [RFC #16](https://github.com/amaranth-lang/rfcs/blob/main/text/0016-soc-csr-regs.md))
+
+#### `csr.reg.Register`
+
+The `csr.reg.Register` class describes a CSR register `Component`, with:
+- a `.__init__(self, fields=None, access=None)` constructor, where:
+    * `fields` is either:
+        - a `dict` that will be instantiated as a `FieldActionMap`;
+        - a `list` that will be instantiated as a `FieldActionArray`;
+        - a `Field` that will be instantiated as a `FieldAction`;
+        - `None`; in this case a `FieldActionMap` is instantiated from `Field` objects in variable annotations.
+    * `access` is either a `csr.Element.Access` value, or `None`.
+- a `.fields` property, returning the instantiated `FieldActionMap`/`FieldActionArray`/`FieldAction`;
+- a `.f` property, as a shorthand to `self.fields`;
+- a `.__iter__(self)` method that yields, for each contained field, a tuple containing its path (as a tuple of names or indices) and its instantiated `FieldAction`. If only a single `Field` was supplied, its path is always `tuple()`.
+
+## Drawbacks
+[drawbacks]: #drawbacks
+
+* A novice might not understand that it's possible to create a `Register` with multiple `Fields` if exposed only to code which uses single-`Field` `Register`s.
+* There is a difference between a `Register` with one named/indexed `Field` and with one unnamed `Field`.
+* Library code will be made more confusing by having `fields` refer to possibly only one field.
+* Downstream code will have to deal with a third type of object returned by the
+  `.f` property which behaves unlike the current two.
+* BSP/doc generators will have to intelligently render a `Field` with an empty name and path (likely by omitting them, possibly by reusing the register's name).
+    * An empty name is prohibited by `FieldActionMap`, so empty names are a novel
+      problem.
+    * The register does not know its own name independently of its container, so this may be difficult in practice.
```

Comment:
No this is not difficult, this is actually how this should be handled to get an actually good BSP generated. This isn't at all a downside, but rather an improvement on the status quo that we can now do this.